core/pom.xml

```diff
@@ -30,6 +30,10 @@
          <artifactId>jgroups</artifactId>
       </dependency>
 
+      <dependency>
```

Comment:
Yes, the user needs to add explicitly the dependency, that's how optional works...
OTOH, adding it as a compile dependency will bring it to the classpath of users who don't care about kubernetes.